interface/rest/rest.go

```diff
@@ -19,7 +19,7 @@ func NewRestAPI(
 ) (*RestAPI, error) {
 	router := mux.NewRouter()
 
-	router.HandleFunc("/shops", NewShopsHandler(searchShopService).Search).Methods(http.MethodGet)
+	router.HandleFunc("/search/shops", NewShopsHandler(searchShopService).Search).Methods(http.MethodGet)
```

Comment:
I guess it's a good time to start thinking about how you want to map out your endpoints. To me, the RESTful way to do this would be to simply have this as:

GET /shops, with optional search parameters, passed via query string. You will probably want to paginate and limit this endpoint, and provide some sort of default functionality here if no search is present. For example, you could show the 50 most popular shops, 50 newest shops, etc.

Reply:
Hm, good point. I could also return the 50 most closest shops to you if no search query is present... But the behaviour would be definitely a bit weird, so I thought of just making this the "search" endpoint and maybe a specific one for just getting the shops which has less "side-effects" and is more RESTful (list all endpoints and maybe filter by newest, closest, ...). WDYT?

Reply:
Yeah, maybe it should have a sort parameter too. But if you want it to be free of "side effects" I guess you could just return them in chronological order, which is what your database is likely to do. Then from your UI, you can just send a default sort order parameter of "closest." Seems legit.

Reply:
I changed it, you’re right with not having two of them...